operator/pkg/helmreconciler/prune.go

```diff
@@ -146,7 +259,12 @@ func (h *HelmReconciler) pruneUnlistedResources(excluded map[string]bool, coreLa
 
 		err := h.client.Delete(context.TODO(), &o, client.PropagationPolicy(metav1.DeletePropagationBackground))
 		if err != nil {
-			errs = util.AppendErr(errs, err)
+			if !strings.Contains(err.Error(), "not found") {
+				errs = util.AppendErr(errs, err)
+			} else {
+				// do not return error if resources are not found
+				h.opts.Log.LogAndPrintf("object: %s is not being deleted: %v", obj.Hash(), err)
```

Comment:
why pass an internal error to the user when you can clearly inform them of the reason? you could just say here "%s will not be deleted because it no longer exists" and that would be a clean, crystal clear message to the user.